File: cpp/diagonal_negativos.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int n;


    cout << "Qual a ordem da matriz? ";
    cin >> n;

    int mat[n][n];
    int negativos = 0;

    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> mat[i][j];
            if (mat[i][j] < 0)
            {
                negativos++;
            }
        }
    }

    cout << "DIAGONAL PRINCIPAL:" << endl;

    for (int i = 0; i < n; i++)
    {
        cout << mat[i][i] << " ";
    }
    cout << "\n";
    cout << "QUANTIDADE DE NEGATIVOS = " << negativos << endl;

    return 0;
}

File: python/diagonal_negativos.py
n: int

negativos: int = 0

n = int(input("Qual a ordem da matriz? "))

mat:  [[int]] = [[0 for x in range(n)] for x in range(n)]

for i in range(0, n):
    for j in range(0, n):
        mat[i][j] = int(input(f"Elemento [{i},{j}]: "))
        if mat[i][j] < 0:
            negativos = negativos + 1
print()
print("DIAGONAL PRINCIPAL:")
for i in range(0, n):
    print(f"{mat[i][i]} ", end="")
print()
print(f"QUANTIDADE DE NEGATIVOS = {negativos}")
